src/Modules/OrchardCore.Commerce/Services/ProductInventoryService.cs

```diff
@@ -25,4 +34,33 @@ public async Task<IList<ShoppingCartItem>> UpdateInventoriesAsync(IList<Shopping
 
         return items;
     }
+
+    public Task<bool> VerifyLinesAsync(IList<ShoppingCartLineViewModel> lines)
+    {
+        var cannotCheckout = false;
+        foreach (var line in lines)
+        {
+            var productPart = line.Product.ContentItem.As<ProductPart>();
+            if (productPart.As<InventoryPart>() is not { } inventoryPart)
+            {
+                continue;
+            }
+
+            var item = new ShoppingCartItem(line.Quantity, line.ProductSku, line.Attributes?.Values);
+            var fullSku = _productService.GetOrderFullSku(item, productPart);
+            var inventoryIdentifier = string.IsNullOrEmpty(fullSku) ? productPart.Sku : fullSku;
+            var relevantInventory = inventoryPart.Inventory.FirstOrDefault(entry => entry.Key == inventoryIdentifier);
+
+            cannotCheckout = relevantInventory.Value < 1 &&
+                !inventoryPart.AllowsBackOrder.Value &&
+                !inventoryPart.IgnoreInventory.Value;
+
+            if (cannotCheckout)
+            {
+                return Task.FromResult(true);
+            }
+        }
+
+        return Task.FromResult(cannotCheckout);
```

Comment:
This will always be false, so it would be better for readability to spell it out here.
(then the `var cannotCheckout` can be moved inside the loop)